### ISSUE TITLE
Truncates some GL extensions when there is a lot of extensions

### ISSUE THREAD

**Issue by baryluk** [ioquake/ioq3#478]
I think, there is a fixed max size, in source code. Not sure if this affects the actual usage, but in the terminal, I see the list is truncated.

From what I see, the buffer that the extensions are added to is statically max 4KB.

**Comment by ensiform** [comment]
The full list known by the engine is not truncated. Only the list that is shared with the mod code for compatibility purposes with qvms.

**Comment by zturtleman** [comment]
OpenGL versions less than 3.0 are [limited to printing 4KB extensions string](https://github.com/ioquake/ioq3/blob/05180e32dcfb9a4552e1b9652b56127248a9950c/code/renderergl1/tr_init.c#L932) though this does not affect usage.

History cause why not: Stock Quake 3 and ioquake3 1.36 (2009)—aka the release on the website—truncate OpenGL extensions displayed in console to 1KB. 3dfd2dac fixes only printing 1KB out of 4KB of extensions string, 06b47ad2 checks for extensions in >4KB string, and d549b64 (opengl2) / c9d12aa (opengl1) makes OpenGL 3.0+ contexts print >4KB string.

**Comment by baryluk** [comment]
Yeah. I believe the all requested extensions do work just fine, it is just printing to the console / terminal that is somehow broken.

**Comment by Calinou** [comment]
Another fun fact: Mesa provides an environment variable to trim the exposed OpenGL extension string list as some old games will crash due to long extension strings.

**Comment by ensiform** [comment]
Nvidia drivers have workaround by exe name lol
